### DISCUSSION TITLE
In Keycloak how can i get the roles for a user?

### DISCUSSION THREAD

**Discussion by felix-exon** [SocialiteProviders/Providers#1033] [Q&A]
My scope is always `openid email profile` which does not include any roles.

How can I change the scope ?

```php
return Socialite::driver('keycloak')
            ->scopes(['openid', 'profile', 'email', 'roles'])
            ->redirect();
```

This does not work with my version:

```json
"laravel/socialite": "^5.6",
 "socialiteproviders/keycloak": "^5.3.0",
```

**Comment by mvdschot** [comment]
Hi,
In my setup on KeyCloak, I've made a Mapper on the Client to make this work:
<img width="1071" alt="image" src="https://github.com/SocialiteProviders/Providers/assets/13979036/df217e62-69b2-4c64-bd09-cf581271012a">
